src/analysis/retail/evoker/preservation/modules/talents/CycleOfLife.tsx

```diff
@@ -0,0 +1,223 @@
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { CYCLE_OF_LIFE_PERCENT_SAVED, ECHO_HEALS, SPELL_COLORS } from '../../constants';
+
+const TRACKED_SPELL_IDS = [
+  SPELLS.CYCLE_OF_LIFE_HEAL.id,
+  SPELLS.DREAM_BREATH.id,
+  TALENTS_EVOKER.ECHO_TALENT.id,
+  SPELLS.EMERALD_BLOSSOM.id,
+  SPELLS.EMERALD_COMMUNION_ALLY.id,
+  SPELLS.EMERALD_COMMUNION_SELF.id,
+  SPELLS.FLUTTERING_SEEDLINGS_HEAL.id,
+  SPELLS.LIFEBIND_HEAL.id,
+  SPELLS.LIVING_FLAME_HEAL.id,
+  SPELLS.RENEWING_BLAZE_HEAL.id,
+  TALENTS_EVOKER.REVERSION_TALENT.id,
+].concat(
+  ECHO_HEALS.map((info) => {
+    return info.id;
+  }),
+);
+
+class CycleOfLife extends Analyzer {
+  savedBySpell: Map<number, number> = new Map<number, number>();
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.CYCLE_OF_LIFE_TALENT);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.handleHeal);
+  }
+
+  handleHeal(event: HealEvent) {
+    if (
+      !this.selectedCombatant.hasBuff(SPELLS.CYCLE_OF_LIFE_BUFF.id) &&
+      !this.selectedCombatant.hasBuff(SPELLS.CYCLE_OF_LIFE_STACK.id)
+    ) {
+      return;
+    }
+    const curVal = this.savedBySpell.has(event.ability.guid)
+      ? this.savedBySpell.get(event.ability.guid)
+      : 0;
+    const amount = event.amount + (event.absorbed || 0) + (event.overheal || 0);
+    const percentSaved = this.selectedCombatant.hasBuff(SPELLS.CYCLE_OF_LIFE_BUFF.id)
+      ? CYCLE_OF_LIFE_PERCENT_SAVED
+      : CYCLE_OF_LIFE_PERCENT_SAVED *
+        this.selectedCombatant.getBuffStacks(SPELLS.CYCLE_OF_LIFE_STACK.id);
+    this.savedBySpell.set(event.ability.guid, curVal! + amount * percentSaved);
+  }
+
+  getOtherHealing(items: Item[]) {
+    let totalHealing = 0;
+    this.savedBySpell.forEach((value: number, key: number) => {
+      if (!TRACKED_SPELL_IDS.includes(key)) {
+        totalHealing += value;
+      }
+    });
+    const allVals = items.sort((a, b) => (a.value > b.value ? 1 : -1));
+    const topVals = new Set<Item>(allVals.slice(-5));
+    for (let i = items.length - 1; i >= 0; i -= 1) {
+      if (!topVals.has(items[i])) {
+        totalHealing += items[i].value;
+        items.splice(i, 1);
+      }
+    }
+    return totalHealing;
+  }
+
+  totalSavedBySpell(spellId: number) {
+    return this.savedBySpell.get(spellId) || 0;
+  }
+
+  renderDonutChart() {
+    const items = [
+      {
+        color: '#fafad2',
+        label: 'Cycle of Life',
+        spellId: TALENTS_EVOKER.CYCLE_OF_LIFE_TALENT.id,
+        value: this.totalSavedBySpell(SPELLS.CYCLE_OF_LIFE_HEAL.id),
+        valueTooltip: formatNumber(this.totalSavedBySpell(SPELLS.CYCLE_OF_LIFE_HEAL.id)),
+      },
+      {
+        color: SPELL_COLORS.DREAM_BREATH,
+        label: 'Dream Breath',
+        spellId: TALENTS_EVOKER.DREAM_BREATH_TALENT.id,
+        value: this.totalSavedBySpell(SPELLS.DREAM_BREATH_ECHO.id + SPELLS.DREAM_BREATH.id),
```

Comment:
`this.totalSavedBySpell(SPELLS.DREAM_BREATH_ECHO.id) + this.totalSavedBySpell(SPELLS.DREAM_BREATH_ECHO.id)`

Reply:
Oops ya